torch/csrc/jit/interpreter.cpp

```diff
@@ -218,6 +230,31 @@ Operation getOperation(jit::Node *node) {
     return [](const list_of_retainable & inputs, list_of_retainable & outputs) {
       outputs.push_back(toRetainableSteal(at::Tensor()));
     };
+  IR_ELSEIF(ReplaceIfUndef)
+    return [](const list_of_retainable & inputs, list_of_retainable & outputs) {
+      auto result = inputs[0];
+      //TODO: refcounting stuff here is ugly but TensorTemporary is not
+      //present. Consider whether we
+      // 1. expose tensor temporary here
+      // 2. keep as is
+      // 3. remove all of this retainable stuff anyway since the new
+      // execution paths do not need handle types.
+      // Note that list_of_retainable is painful because it is yet another
+      // list of pointers that require needless copies.
+      if(result == at::UndefinedTensor::singleton()) {
+        result = inputs[1];
+      }
+      result->retain();
+      outputs.push_back(result);
```

Comment:
Does `list_of_retainable` steal by default?

Reply:
`inputs` are borrowed. Anything in `outputs` should be +1. This stuff needs to change anyway to enable handing off ownship of tensors to subgraphs so I am not too worried about this code lasting a long time. I am trying to figure out to re-work the interpreter to enable this and hopefully make the list_of_retainables more straightforward.